### ISSUE TITLE
PyTorch 1.1 release

### ISSUE THREAD

**Issue by BenjaminBossan** [skorch-dev/skorch#467]
There has been a new PyTorch release ([notes](https://github.com/pytorch/pytorch/releases)) and we should update skorch to keep pace. On my notebook (no GPU), all but 3 tests passed. These one failed:
https://github.com/skorch-dev/skorch/blob/17256a1325ea463003a0e927a09a11065dc0b21a/skorch/tests/test_net.py#L313-L322
This one could probably be removed, since it tests whether we can unpickle an old state (<=0.5).

https://github.com/skorch-dev/skorch/blob/17256a1325ea463003a0e927a09a11065dc0b21a/skorch/tests/callbacks/test_lr_scheduler.py#L22-L27
The test fails for both parametrizations. Since this uses `LRScheduler`, it might have to do with the breaking change introduced [here](https://github.com/pytorch/pytorch/pull/7889).

I will also run the tests on GPU once I have time.

**Comment by thomasjpfan** [comment]
The change in https://github.com/pytorch/pytorch/pull/7889 changes the order of calling `step` in the scheduler. Looking at the [StepLR](https://pytorch.org/docs/stable/optim.html?highlight=steplr#torch.optim.lr_scheduler.StepLR) (and the other schedulers) documentation, it suggest to run `step` at the end of a epoch now.

**Comment by thomasjpfan** [comment]
The fix to `simulate` would most likely break on other versions of PyTorch. We would also need to move the `step` call to `on_epoch_end` which will change behavior for users on PyTorch 1.0.

**Comment by BenjaminBossan** [comment]
No, let's not have code that branches based on PyTorch version. If we made a change to `StepLR` and at the same time increase the minimum PyTorch version to 1.1.0, would that result in the same behavior as before? If so, I would argue that it's not a breaking change.

**Comment by thomasjpfan** [comment]
> If we made a change to StepLR and at the same time increase the minimum PyTorch version to 1.1.0, would that result in the same behavior as before?

Yes, as long as we dump the minimum version, everything should work as before. With this I will make a PR to fix the scheduler part.

On a side note, we can run our tests nightly on TravisCI with the nightly version of PyTorch. This way we can better prepare for a new PyTorch release.

**Comment by BenjaminBossan** [comment]
> Yes, as long as we dump the minimum version, everything should work as before. With this I will make a PR to fix the scheduler part.

Sounds good.

> On a side note, we can run our tests nightly on TravisCI with the nightly version of PyTorch. This way we can better prepare for a new PyTorch release.

That sounds good, as long as it doesn't affect our badge ;)

A small update (or rather non-update) on testing PyTorch 1.1: My desktop decided to become a brick and as I have no time right now to fix this, I cannot test with cuda at the moment.

**Comment by ottonemo** [comment]
> A small update (or rather non-update) on testing PyTorch 1.1: My desktop decided to become a brick and as I have no time right now to fix this, I cannot test with cuda at the moment.

Ugh, that's unfortunate. I ran the tests on a CUDA device and they work fine. I get the same errors as you did.

>  This one could probably be removed, since it tests whether we can unpickle an old state (<=0.5).

Either that or we generate a new pickle from version 0.5.0 with PyTorch 1.1.0. I opt for the latter since we haven't released 0.5.1 yet :)

**Comment by BenjaminBossan** [comment]
Yeah, with that change to PyTorch 1.1.0, we should also make a new release.

**Comment by BenjaminBossan** [comment]
After the fixes introduced by #473, only [this test](https://github.com/skorch-dev/skorch/blob/master/skorch/tests/test_net.py#L312-L322) is still failing. The reason is that the following line has been added: https://github.com/pytorch/pytorch/blame/6e82b1c77d36386ba738af3287693105b4bbafe2/torch/optim/optimizer.py#L54.

This is to be expected. I see no easy fix to this problem -- basically, all pickle files (that use optimizers) will stop working with PyTorch 1.1. The way forward is probably to accept this, that is just a risk that comes with pickle. If anyone has an idea for an easy fix, I'm open to suggestions.

**Comment by thomasjpfan** [comment]
When users are deploying a model, I hope they fix their dependencies so they don’t run into a pickling issue. 

I’m +1 for accepting this pickling issue.

**Comment by ottonemo** [comment]
This has been fixed in #476.
